src/Fixer/PhpUnit/PhpUnitTestCaseStaticMethodCallsFixer.php

```diff
@@ -0,0 +1,447 @@
+            if (isset($this->configuration['methods'][$methodName])) {
+                $callType = $this->configuration['methods'][$methodName];
+            }
+
+            $operatorIndex = $tokens->getPrevMeaningfulToken($index);
+            $referenceIndex = $tokens->getPrevMeaningfulToken($operatorIndex);
+            if (!$this->needsConversion($tokens, $operatorIndex, $referenceIndex, $callType)) {
+                continue;
+            }
+
+            if (self::CALL_TYPE_THIS === $callType && $this->isInsideStaticFunction($tokens, $analyzer, $index)) {
+                continue;
+            }
+
+            $tokens[$operatorIndex] = new Token($this->conversionMap[$callType][0]);
+            $tokens[$referenceIndex] = new Token($this->conversionMap[$callType][1]);
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $operatorIndex
+     * @param int    $referenceIndex
+     * @param string $callType
+     *
+     * @return bool
+     */
+    private function needsConversion(Tokens $tokens, $operatorIndex, $referenceIndex, $callType)
+    {
+        if ($tokens[$operatorIndex]->equals([T_DOUBLE_COLON, '::']) && $tokens[$referenceIndex]->equals([T_STATIC, 'static'])) {
+            return self::CALL_TYPE_STATIC !== $callType;
+        }
+
+        if ($tokens[$operatorIndex]->equals([T_OBJECT_OPERATOR, '->']) && $tokens[$referenceIndex]->equals([T_VARIABLE, '$this'])) {
+            return self::CALL_TYPE_THIS !== $callType;
+        }
+
+        if ($tokens[$operatorIndex]->equals([T_DOUBLE_COLON, '::']) && $tokens[$referenceIndex]->equals([T_STRING, 'self'])) {
+            return self::CALL_TYPE_SELF !== $callType;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param Tokens         $tokens
+     * @param TokensAnalyzer $analyzer
+     * @param int            $index
+     *
+     * @return bool
+     */
+    private function isInsideStaticFunction(Tokens $tokens, TokensAnalyzer $analyzer, $index)
+    {
+        $functionIndex = $tokens->getPrevTokenOfKind($index, [[T_FUNCTION]]);
+        while ($analyzer->isLambda($functionIndex)) {
+            $openingCurlyBraceIndex = $tokens->getNextTokenOfKind($functionIndex, ['{']);
+            $closingCurlyBraceIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $openingCurlyBraceIndex);
+
+            if ($closingCurlyBraceIndex > $index) {
+                $prev = $tokens->getPrevMeaningfulToken($functionIndex);
+                if ($tokens[$prev]->isGivenKind(T_STATIC)) {
+                    return true;
+                }
+            }
+
+            $functionIndex = $tokens->getPrevTokenOfKind($functionIndex, [[T_FUNCTION]]);
+        }
+
+        $prev = $tokens->getPrevMeaningfulToken($functionIndex);
+        if (!$tokens[$prev]->isGivenKind(T_STATIC)) {
+            $prev = $tokens->getPrevMeaningfulToken($prev);
+        }
+
+        return $tokens[$prev]->isGivenKind(T_STATIC);
```

Comment:
If the function signature is `static final private function ...`, this will return `false`.

Reply:
Thank you, fixed